diff --git a/cpan/Encode/Byte/Makefile.PL b/cpan/Encode/Byte/Makefile.PL
index 6824bbb..714930e 100644
--- a/cpan/Encode/Byte/Makefile.PL
+++ b/cpan/Encode/Byte/Makefile.PL
@@ -1,7 +1,7 @@
 use 5.7.2;
 use strict;
 use ExtUtils::MakeMaker;
-use File::Spec::Functions;
+use File::Spec;
 
 my $name = 'Byte';
 my %tables = (
@@ -44,15 +44,6 @@ my %not_here =
  qw(cp932.ucm cp936.ucm cp949.ucm cp950.ucm),  # CJK
  );
 
-opendir(ENC,catdir(updir(),'ucm')) or die $!;
-while (defined(my $file = readdir(ENC)))
-{
-    $file =~ /^(8859|cp).*\.ucm$/io or next;
-    $not_here{$file} and next;
-    push(@{$tables{byte_t}},$file);
-}
-closedir(ENC);
-
 WriteMakefile(
               INC		=> "-I../Encode",
           NAME		=> 'Encode::'.$name,
