--- xorg-server-21.1.18.orig/hw/xfree86/fbdevhw/fbdevhw.c	2026-04-06 20:30:54.000000000 +0000
+++ xorg-server-21.1.18/hw/xfree86/fbdevhw/fbdevhw.c	2026-04-06 20:34:15.000000000 +0000
@@ -329,6 +329,7 @@
         return -1;
     }
 
+#ifndef __emerixx__
     /* only touch non-PCI devices on this path */
     {
         char buf[PATH_MAX] = {0};
@@ -344,6 +345,7 @@
         }
         free(sysfs_path);
     }
+#endif
 
     if (namep) {
         if (-1 == ioctl(fd, FBIOGET_FSCREENINFO, (void *) (&fix))) {
--- xorg-server-21.1.18.orig/hw/xfree86/common/xf86Bus.c	2026-04-06 22:29:24.000000000 +0000
+++ xorg-server-21.1.18/hw/xfree86/common/xf86Bus.c	2026-04-06 22:30:37.000000000 +0000
@@ -556,6 +556,8 @@
 void
 xf86PostProbe(void)
 {
+#if !defined(__emerixx__)
+  
     if (fbSlotClaimed && (
 #if (defined(__sparc__) || defined(__sparc)) && !defined(__OpenBSD__)
                              sbusSlotClaimed ||
@@ -571,6 +573,7 @@
         ))
         FatalError("Cannot run in framebuffer mode. Please specify busIDs "
                    "       for all framebuffer devices\n");
+#endif
 }
 
 Bool
