diff --git a/Makefile b/Makefile
index a231ae8..3cf164a 100644
--- a/Makefile
+++ b/Makefile
@@ -19,28 +19,20 @@
 #
 
 HAVE_GZIP		= 1
-HAVE_LINUX_ATTR		= 1
-HAVE_LINUX_CHROOT	= 1
-HAVE_LINUX_LOSETUP	= 1
-HAVE_LINUX_PIVOT	= 1
-HAVE_LINUX_MOUNT	= 1
+HAVE_LINUX_ATTR		= 0
+HAVE_LINUX_CHROOT	= 0
+HAVE_LINUX_LOSETUP	= 0
+HAVE_LINUX_PIVOT	= 0
+HAVE_LINUX_MOUNT	= 0
 HAVE_BSD_MOUNT		= 0
 MOUNT_TYPE		= '"ext3"'
 
 OPT = -O3
 
-CFLAGS = $(OPT) -Wall -Wmissing-prototypes \
-	-DHAVE_GZIP=$(HAVE_GZIP) \
-	-DHAVE_LINUX_ATTR=$(HAVE_LINUX_ATTR) \
- 	-DHAVE_LINUX_CHROOT=$(HAVE_LINUX_CHROOT) \
- 	-DHAVE_LINUX_LOSETUP=$(HAVE_LINUX_LOSETUP) \
- 	-DHAVE_LINUX_PIVOT=$(HAVE_LINUX_PIVOT) \
-	-DHAVE_LINUX_MOUNT=$(HAVE_LINUX_MOUNT) \
-	-DHAVE_BSD_MOUNT=$(HAVE_BSD_MOUNT) \
-	-DMOUNT_TYPE=$(MOUNT_TYPE)
+CFLAGS = $(OPT) -Wall -Wmissing-prototypes
 
 LDFLAGS = -static
-LIBS = -lz
+LIBS =
 
 
 DESTDIR =
diff --git a/cmds.c b/cmds.c
index 8e530db..ce18129 100644
--- a/cmds.c
+++ b/cmds.c
@@ -28,7 +28,7 @@
 #else
 #define dev_t unsigned short
 #endif
-#include <linux/loop.h>
+//#include <linux/loop.h>
 #undef dev_t
 #define dev_t dev_t
 
@@ -667,9 +667,9 @@ do_mount(int argc, const char ** argv)
 	argc--;
 	argv++;
 
+	
+	#if	HAVE_LINUX_MOUNT
 	type = MOUNT_TYPE;
-
-#if	HAVE_LINUX_MOUNT
 	flags = MS_MGC_VAL;
 #else
 	flags = 0;
